File: src/services/driverService.js
import api from './api'
export default {
    getAll() {
        return api.get('/driver/getAll')
    },
    create(driver) {
        return api.post('/driver/create', driver)
    },
    read(id) {
        return api.get(`/driver/read/${id}`)
    },
    update(driver) {
        return api.put('/driver/update', driver)
    },
    delete(id) {
        return api.delete(`/driver/delete/${id}`)
    }
}

File: src/main/java/za/ac/cput/service/DriverService.java
package za.ac.cput.service;
/*
DriverServiceImpl.java
Driver service implementation
Author: Angel Dineo Masonganye (223008869)
Date: 2026
*/
import org.springframework.stereotype.Service;
import za.ac.cput.domain.Driver;
import za.ac.cput.repository.DriverRepo.IDriverRepository;

import java.util.List;

@Service
public class DriverService implements IDriverService {
    private final IDriverRepository repository;


    public DriverService(IDriverRepository repository) {
        this.repository = repository;
    }

    @Override
    public Driver create(Driver driver) {
        return repository.save(driver);
    }

    @Override
    public Driver read(String id) {
        return repository.findById(id).orElse(null);
    }

    @Override
    public Driver update(Driver driver) {
        if (repository.existsById(driver.getId())) {
            return repository.save(driver);
        }
        return null;
    }

    @Override
    public boolean delete(String id) {
        if (repository.existsById(id)) {
            repository.deleteById(id);
            return true;
        }
        return false;
    }

    @Override
    public List<Driver> getAll() {
        return repository.findAll();
    }
}
